support_files/patches/180214_fix_upper_case.py

```diff
@@ -0,0 +1,56 @@
+# run within the Qiita environment, suggest using a screen session
+
+from qiita_db.study import Study
+from qiita_db.software import Software
+from biom import load_table
+from qiita_db.util import compute_checksum
+from qiita_db.sql_connection import TRN
+from biom.util import biom_open
+from skbio.io import read
+from os import rename, remove
+
+
+studies = Study.get_by_status('private').union(
+    Study.get_by_status('public')).union(Study.get_by_status('sandbox'))
+
+sft = Software.from_name_and_version('deblur', '1.0.4')
+sft = Software.from_name_and_version('deblur', '1.0.3')
+# [0] deblur only has 1 command
+cmd = sft.commands[0]
+
+artifacts = [a for s in studies for a in s.artifacts()
+             if a.processing_parameters is not None
+             and a.processing_parameters.command == cmd]
+
+sql = "UPDATE qiita.filepath SET checksum = %s WHERE filepath_id = %s"
+for a in artifacts:
+    # putting all this in a transaction in case something fails it does
+    # it nicely
+    with TRN:
+        ftps = {ft: (fid, fp) for fid, fp, ft in a.filepaths
+                if ft in ['biom', 'preprocessed_fasta']}
+        biom = ftps['biom'][1]
+        t = load_table(biom)
+        current = t.ids('observation')
+        updated = map(lambda x: x.upper(), current)
+        if set(current) ^ set(updated):
+            print '*********>\nChanging biom: ', a.id, fp
+                t.update_ids({i: i.upper() for i in t.ids('observation')},
```

Comment:
I am not familiar enough with biom, but I think you can encounter a situation where you have two observations like 'ACGt' and 'ACGT' that were two separate rows in the old table, but now need to be collapsed, i.e. their read counts for the samples need to be summed, to the single observation 'ACGT'.
Does you script account for this necessary collapsing of counts?

Reply:
The code doesn't account for that but the last time we discussed this, we believe that in a single biom (from a single preprocessing) it's not possible to get the same sequence with different casing, cc-ing: @ElDeveloper the main source of this information. Note that is possible that in a meta-analysis the same sequence can be found with different casing.

Reply:
Theoretically, I agree. But I think it wouldn't hurt if we compare the set size difference between current and updated. It there is no difference - as expected - good, but if there is we don't want to apply this to our life data! Good that we have the RC version where we can execute it first.

Reply:
@rnaer, I think you mentioned you observed this the other day?

Reply:
> ... we believe that in a single biom (from a single preprocessing) it's not possible to get the same sequence with different casing

hmmm I don't think we have any guarantee that this won't happen. I think we just assumed that deblur wouldn't do this. So it's not really a guarantee.

Reply:
I saw the same sequence in 2 observations like 'ACGt' and 'ACGT' in the latest AGP biom table - but I got it from Austin and don't know if this table is from single processing or merged.

Reply:
OK, I'll add some code for this.